src/printer-python.js

```diff
@@ -0,0 +1,538 @@
+    }
+
+    case "Dict": {
+      const keys = path.map(print, "keys");
+      const values = path.map(print, "values");
+
+      const pairs = keys.map((k, i) => concat([softline, k, ": ", values[i]]));
+
+      return concat(["{", indent(join(",", pairs)), softline, "}"]);
+    }
+
+    case "ClassDef": {
+      let bases = [];
+
+      if (n.bases.length > 0) {
+        bases = ["(", join(",", path.map(print, "bases")), ")"];
+      }
+
+      return concat([
+        "class ",
+        n.name,
+        concat(bases),
+        ":",
+        indent(concat([hardline, printBody(path, print)]))
+      ]);
+    }
+
+    case "Attribute": {
+      return concat([path.call(print, "value"), ".", n.attr]);
+    }
+
+    case "Add": {
+      return "+";
+    }
+
+    case "USub":
+    case "Sub": {
+      return "-";
+    }
+
+    case "Mult": {
+      return "*";
+    }
+
+    case "MatMult": {
+      return "@";
+    }
+
+    case "Div": {
+      return "/";
+    }
+
+    case "FloorDiv": {
+      return "//";
+    }
+
+    case "Mod": {
+      return "%";
+    }
+
+    case "Pow": {
+      return "**";
+    }
+
+    case "LShift": {
+      return "<<";
+    }
+
+    case "RShift": {
+      return ">>";
+    }
+
+    case "BitAnd": {
+      return "&";
+    }
+
+    case "BitXor": {
+      return "^";
+    }
+
+    case "BitOr": {
+      return "|";
+    }
+
+    case "Compare": {
+      const ops = path.map(print, "ops");
+      const comparators = path.map(print, "comparators");
+
+      const pairs = ops.map((op, i) => concat([" ", op, " ", comparators[i]]));
+
+      return concat([path.call(print, "left"), ...pairs]);
+    }
+
+    case "Lt": {
+      return "<";
+    }
+
+    case "LtE": {
+      return "<=";
+    }
+
+    case "Gt": {
+      return ">";
+    }
+
+    case "GtE": {
+      return ">=";
+    }
+
+    case "Eq": {
+      return "==";
+    }
+
+    case "NotEq": {
+      return "!=";
+    }
+
+    case "Import": {
+      return concat(["import ", join(", ", path.map(print, "names"))]);
+    }
+
+    case "ImportFrom": {
+      return concat([
+        "from ",
+        ".".repeat(n.level),
+        n.module,
+        " import ",
+        join(", ", path.map(print, "names"))
+      ]);
+    }
+
+    case "alias": {
+      if (n.asname) {
+        return `${n.name} as ${n.asname}`;
+      }
+
+      return n.name;
+    }
+
+    case "While": {
+      return concat([
+        "while ",
+        path.call(print, "test"),
+        ":",
+        indent(concat([hardline, printBody(path, print)]))
+      ]);
+    }
+
+    case "If": {
+      let ifType = "if ";
+
+      if (path.getParentNode().ast_type === "If") {
+        ifType = "elif ";
+      }
+
+      let parts = [
+        ifType,
+        path.call(print, "test"),
+        ":",
+        indent(concat([hardline, printBody(path, print)]))
+      ];
+
+      if (n.orelse.length > 0) {
+        if (n.orelse.length === 1 && n.orelse[0].ast_type !== "If") {
+          parts = [
```

Comment:
parts.push(line, "else:", indent(concat([line, concat(path.map(print, "orelse"))])))

Reply:
I can't believe I didn't know that `push` accepted multiple parameters! All fixed now :)